src/foundry/foundry.js/roll.d.ts

```diff
@@ -1,501 +1,505 @@
-   * @param missing - The value that should be assigned to any unmatched keys.
-   *                  If null, the unmatched key is left as-is.
-   * @param warn    - Display a warning notification when encountering an un-matched key.
-   *                  (default: `false`)
-   */
-  static replaceFormulaData<D>(
-    formula: string,
-    data: D,
-    {
-      missing,
-      warn
-    }?: {
-      missing?: string;
-      warn?: boolean;
-    }
-  ): string;
+    /**
+     * Create a new Roll object using the original provided formula and data.
+     * Each roll is immutable, so this method returns a new Roll instance using the same data.
+     * @param options - Evaluation options passed to Roll#evaluate
+     * @returns A new Roll object, rolled using the same formula and data
+     */
+    reroll(options?: Partial<Options & { async: false }>): this;
+    reroll(options?: Partial<Options & { async: true }>): Promise<this>;
+    reroll(options?: Partial<Options>): this | Promise<this>;
 
-  /**
-   * Split a formula by identifying its outer-most parenthetical and math terms
-   * @param _formula - The raw formula to split
-   * @returns An array of terms, split on parenthetical terms
-   */
-  protected static _splitParentheses(_formula: string): string[];
+    /**
+     * A factory method which constructs a Roll instance using the default configured Roll class.
+     * @typeParam D - the type of data object against which to parse attributes within the formula
+     * @param formula - The formula used to create the Roll instance
+     * @param data    - The data object which provides component data for the formula
+     * @param options - Additional options which modify or describe this Roll
+     * @returns The constructed Roll instance
+     */
+    static create<D extends Record<string, unknown> = {}>(
+      formula: string,
+      data?: D,
+      options?: Partial<Options>
+    ): Roll<D>;
 
-  /**
-   * Handle closing of a parenthetical term to create a MathTerm expression with a function and arguments
-   */
-  protected static _splitMathTerm(fn: Function, term: string): MathTerm[];
+    /**
+     * Transform an array of RollTerm objects into a cleaned string formula representation.
+     * @param terms - An array of terms to represent as a formula
+     * @returns The string representation of the formula
+     */
+    static getFormula(terms: RollTerm[]): string;
 
-  /**
-   * Split a formula by identifying its outer-most dice pool terms
-   * @param _formula - The raw formula to split
-   * @returns An array of terms, split on parenthetical terms
-   */
-  protected _splitPools(_formula: string): string[];
+    /**
+     * A sandbox-safe evaluation function to execute user-input code with access to scoped Math methods.
+     * @param expression - The input string expression
+     * @returns The numeric evaluated result
+     */
+    static safeEval(expression: string): number | unknown;
```

Comment:
`number | unknown === unknown`. But why do you add the type `unknown` here? Looking at the function, it seems like it will always return a `number` or throw an Error...?